modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase

```diff
@@ -3065,6 +3065,101 @@ definition {
 		}
 	}
 
+	@description = "LPS-180195 - Verify on the Account object if it's possible to trigger an action with ADD when ADD an account"
+	@priority = 5
+	test CanTriggerAddActionWithSystemObjectToItselfWhenAdd {
+		property portal.acceptance = "true";
+
+		task ("Given a Account object and given an action to create a new Account entry after adding an Account entry") {
+			ObjectAdmin.addObjectViaAPI(
+				labelName = "Custom Object 180195",
+				objectName = "CustomObject180195",
+				pluralLabelName = "Custom Objects 180195");
+
+			ObjectAdmin.addObjectFieldViaAPI(
+				fieldBusinessType = "Integer",
+				fieldLabelName = "Custom Field",
+				fieldName = "customField",
+				fieldType = "Integer",
+				isRequired = "false",
+				objectName = "CustomObject180195");
+
+			ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject180195");
+		}
+
+		task ("When the user adds a new Account entry") {
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object 180195");
+
+			ObjectAdmin.goToActionsTab();
+
+			ObjectAdmin.gotoAddObjectAction();
+
+			Type(
```

Comment:
Macros please for type and clicks. Having the actions here does not describe what's going on and it makes it harder to understand the flow of the test. Other team members or test analysts will not be able to readily decode what's going on here. 

Also, macros for reusability.